celery/apps/worker.py

```diff
@@ -141,12 +141,25 @@ def on_start(self):
             app.log.redirect_stdouts(self.redirect_stdouts_level)
 
         # TODO: Remove the following code in Celery 6.0
-        if app.conf.maybe_warn_deprecated_settings():
-            logger.warning(
-                "Please run `celery upgrade settings path/to/settings.py` "
-                "to avoid these warnings and to allow a smoother upgrade "
-                "to Celery 6.0."
-            )
+        # Ok this qualifies as a hack for issue #6366. Tried making it less of
+        # a hack via app.__reduce_keys__(), but that may not work properly in
+        # all cases
+        warn_deprecated = True
+        config_source = app._config_source
+        if isinstance(config_source, (str, )):
```

Comment:
```suggestion
        if isinstance(config_source, str):
```
A tuple is only required if we have multiple types to test `config_source` against.